utils/declaredScope.js

```diff
@@ -4,6 +4,7 @@ exports.__esModule = true;
 
 /** @type {import('./declaredScope').default} */
 exports.default = function declaredScope(context, name) {
+  // todo: what should be done about this...
```

Comment:
can you elaborate more on what "this" is?

Reply:
"this" being "usage of `context.getScope` which was removed in ESLint v9 - plugin-wise I've handled this by introducing a set of utility helpers, but I believe this is being installed as a dedicated package so I decided to just todo it for now while I worked on addressing the more low hanging fruit

Reply:
Perhaps moving the `context` compat layer to the utils package?  That'd allow this function to  use it too (and the plugin would import it from this package?